### ISSUE TITLE
Crash in hot-seat mode after defeating battle

### ISSUE THREAD

**Issue by misiokles** [vcmi/vcmi#2444]
**Describe the bug**
Crash in hot-seat mode after defeating conquered town.

**Game logs**
Save
[testgatekey.zip](https://github.com/vcmi/vcmi/files/12208869/testgatekey.zip)

**To Reproduce**
1. Open testgatekey save in hot-seat mode (2 players - red and blue)
2. Attack with Sanya Wise Oak Town
3. Press autobattle button twice (for every player)
4. Let Sanya loose battle
5. Observe crash by pressing OK at casualties window.

**Expected behavior**
Game shouldn't crash

**Actual behavior**
Game crashes

**Did it work earlier?**
I don't know

**Version**
 - OS:  Windows
 - Version: [VCMI beta develop]

**Additional context**
Used mods: In the Wake of Gods.

**Comment by misiokles** [comment]
Strangelly this piece of commands
```
			{
				"type" : "SPELL_LIKE_ATTACK",
				"subtype" : "spell.infernoGood",
				"val" : 3,
				"limiters": [ {
					"type":"CREATURE_TYPE_LIMITER",
						"parameters": [ "arrowTower", false ] 
					}
				]
			},
```
written in artifact section doesn't work - all shooter can gain inferno ability, not only arrow towers.

**Comment by DjWarmonger** [comment]
Posibly fixed by #2546
